### PR TITLE
feat(integer): add signed_overflowing_scalar_add/sub

### PR DESCRIPTION
**Pull request by tmontaigu**

### CODE REVIEW

File: tfhe/src/integer/server_key/radix_parallel/tests_signed.rs [hunk 1]
```diff
@@ -2989,6 +3003,290 @@ where
...
+                decrypted_overflowed,
+                expected_overflowed,
+                "Invalid overflow flag result for overflowing_add, for ({clear_lhs} + {clear_rhs}) % {modulus} \
+                expected overflow flag {expected_overflowed}, got {decrypted_overflowed}"
+            );
+            assert_eq!(result_overflowed.0.degree.get(), 1);
+            assert_eq!(result_overflowed.0.noise_level(), NoiseLevel::NOMINAL);
+        }
+    }
+
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_signed.rs:3085]: no hardcoded inputs to check that e.g. -modulus + -1 overflows ?

**Review by IceTDrinker — COMMENTED** [review]
Just a question on tests and maybe hardcoded test cases

File: tfhe/src/integer/server_key/radix_parallel/tests_signed.rs [hunk 2]
```diff
@@ -58,8 +59,18 @@ fn signed_add_under_modulus(lhs: i64, rhs: i64, modulus: i64) -> i64 {
...
 fn signed_overflowing_add_under_modulus(lhs: i64, rhs: i64, modulus: i64) -> (i64, bool) {
     assert!(modulus > 0);
-    let mut res = lhs + rhs;
-    let mut overflowed = false;
+    assert!((-modulus..modulus).contains(&lhs));
+
+    // The code below requires rhs and lhs to be in range -modulus..modulus
+    // in scalar tests, rhs may exceed modulus
+    // so we truncate it (is the fhe ops does)
+    let (mut res, mut overflowed) = if (-modulus..modulus).contains(&rhs) {
```
[the remark this answers is not in the document]
Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed.rs:67]: What is supposed to be in range ?
As the comment says, rhs has to be in range for the code after the if to work, but in scalar test,
we intentionnally add rhs that are > modulus so we handle that beforehand

File: tfhe/src/integer/server_key/radix_parallel/tests_signed.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed.rs:3085]: We could add one

File: tfhe/src/integer/server_key/radix_parallel/tests_signed.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed.rs:67]: rhs, sorry, I missed the fact we add one that is intentionally too big

File: tfhe/src/integer/server_key/radix_parallel/tests_signed.rs [hunk 3]
```diff
@@ -2989,6 +3003,342 @@ where
...
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    sks.set_deterministic_pbs_execution(true);
+
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = (cks.parameters().message_modulus().0.pow(NB_CTXT as u32) / 2) as i64;
+
+    let hardcoded_values = [(-modulus, 1), (modulus, -1)];
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_signed.rs:3188]: last question is the biggest value modulus or modulus - 1 ?

Comment (IceTDrinker) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed.rs:3188]: same thing for the other tests

File: tfhe/src/integer/server_key/radix_parallel/tests_signed.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_signed.rs:3201]: adding a hardcoded test of  overflowing_sub(-1, -modulus) allowed to catch a little bug

**Review by IceTDrinker — APPROVED** [review]
Looks good, thanks !

**Pull request merged**